### PR TITLE
Fix part of #3602: Add label for Walk Through Activity

### PR DESCRIPTION
**Pull request by priyatanu**
## Explanation
Fix part of #3602: Add label for Walk Through Activity

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -215,6 +215,7 @@
       android:theme="@style/OppiaThemeWithoutActionBar" />
     <activity
       android:name=".app.walkthrough.WalkthroughActivity"
+      android:label="@string/walk_through_activity_title"
```
Comment (BenHenning) [inline app/src/main/AndroidManifest.xml]: We also need to make sure that there's a test to verify this. See https://github.com/oppia/oppia-android/pull/3227/files as a reference PR for all the parts needed to add a new accessibility label.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @priyatanu! I left one high-level comment. However, beyond that:
- The PR title should be "Fix part of #3602" since this isn't addressing everything covered by that issue
- You should add a description in the explanation section of the PR to explain what this PR is doing
- The "For UI-specific PRs only" portion of the PR description needs to be filled in since this PR is changing a user behavior

Please update these & at-mention me once they've been addressed.

**Comment by priyatanu** [comment]
> Hi @priyatanu, I'm going to mark this PR as stale because it hasn't had any updates for 7 days. If no further activity occurs within **7 days**, it will be automatically closed so that others can take up the issue. If you are still working on this PR, please make a follow-up commit within **3 days** (and submit it for review, if applicable). Please also let us know if you are stuck so we can help you!

Thank you. I will look into this

**Comment by BenHenning** [comment]
FYI I'm listed as a codeowner for this PR & I'll be unavailable to perform code reviews over the next 2 weeks--thanks for your patience.

**Comment by rt4914** [comment]
@ARJUPTA Can you please review this PR first? Thanks.

**Comment by rt4914** [comment]
@priyatanu Thanks for the PR. Please assign PRs to @ARJUPTA first has he your mentor that way its easier to review PRs and communicate effectively.

**Comment by priyatanu** [comment]
> @priyatanu Thanks for the PR. Please assign PRs to @ARJUPTA first has he your mentor that way its easier to review PRs and communicate effectively.

Thank you

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (rt4914) [inline app/src/main/AndroidManifest.xml]: `Walkthrough` is a single word. Rename to
`walkthrough_activity_title`

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -380,6 +380,7 @@
   <!-- CompletedStoryListActivity -->
   <string name="completed_story_list_activity_title">Stories Completed</string>
   <!-- WalkthroughActivity -->
+  <string name="walk_through_activity_title">Walk Through</string>
```
Comment (rt4914) [inline app/src/main/res/values/strings.xml]: `<string name="walkthrough_activity_title">Walkthrough</string>`

@BenHenning What can be more meaningful title here?
E.g.: `App walkthrough` ?

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+import javax.inject.Singleton
+
+/** Tests for [WalkthroughActivityLabelTest]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = WalkthroughActivityLabelTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class WalkthroughActivityLabelTest {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt]: Rename to `WalkthroughActivityTest` here and everywhere else in this file.

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [WalkthroughActivityLabelTest]. */
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt]: Nit: `/** Tests for [WalkthroughActivity]. */`

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+        0
+      )
+    )
+    val title = activityTestRule.activity.title
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    Truth.assertThat(title).isEqualTo(context.getString(R.string.walk_through_activity_title))
+  }
+
+  private fun testWalkThroughActivity_hasCorrectActivityLabel(profileId: Int): Intent {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt]: Rename to `createWalkthroughActivityIntent`

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+  fun testWalkthroughActivity_hasCorrectActivityLabel() {
+    activityTestRule.launchActivity(
+      testWalkThroughActivity_hasCorrectActivityLabel(
+        0
+      )
+    )
+    val title = activityTestRule.activity.title
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    Truth.assertThat(title).isEqualTo(context.getString(R.string.walk_through_activity_title))
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt]: Nit
`assertThat(title).isEqualTo(context.getString(R.string.walk_through_activity_title))`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@priyatanu PTAL.
Also please add screenshot in PR description showing that test is is passing on Espresso.

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt (hunk 3 above)
Comment (priyatanu) [reply app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt]: ok. thanks for the review

Comment (priyatanu) [reply app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt]: Hi, 
There is already a test WalkthroughActivityTest , so I will remove this
class and add a test case to that one

**Comment by priyatanu** [comment]
ExpressoWalkThroughtest

**Comment by priyatanu** [comment]
LandScapre

**Comment by priyatanu** [comment]
Potrait

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt (hunk 6 above)
Comment (priyatanu) [reply app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt (hunk 5 above)
Comment (priyatanu) [reply app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityLabelTest.kt]: Reverted this channge

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (priyatanu) [reply app/src/main/res/values/strings.xml]: Done

**Review by Arjupta — CHANGES_REQUESTED** [review]
@priyatanu one static check is failing because we have previously kept the `WalkthroughActivity` exempted from failing accessiblity issues but as you have resolved them now so it is unnecessary for the file to be exempted further, hence you need to remove the name of `WalkthroughActivity` from `scripts/assets/accessibility_label_exemptions.textproto`

Screenshot 2021-12-08 205741

**Comment by priyatanu** [comment]
> @priyatanu one static check is failing because we have previously kept the `WalkthroughActivity` exempted from failing accessiblity issues but as you have resolved them now so it is unnecessary for the file to be exempted further, hence you need to remove the name of `WalkthroughActivity` from `scripts/assets/accessibility_label_exemptions.textproto`
> 
> Screenshot 2021-12-08 205741



> @priyatanu one static check is failing because we have previously kept the `WalkthroughActivity` exempted from failing accessiblity issues but as you have resolved them now so it is unnecessary for the file to be exempted further, hence you need to remove the name of `WalkthroughActivity` from `scripts/assets/accessibility_label_exemptions.textproto`
> 
> Screenshot 2021-12-08 205741

I have removed the file from the exemptems proto file

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt [hunk 7]
```diff
@@ -173,6 +189,15 @@ class WalkthroughActivityTest {
     }
   }
 
+  @Test
+  fun testWalkthroughActivity_hasCorrectActivityLabel() {
+    val scenario = launch(WalkthroughActivity::class.java)
+    scenario.onActivity { activity ->
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt]: Please write test case similar to `HomeActivityTest` also written in https://github.com/oppia/oppia-android/pull/3850/files

**Review by rt4914 — CHANGES_REQUESTED** [review]
@priyatanu PTAL thanks.

Also, please mention `Fix part of #3602: Add label for Walk Through Activity` in your PR description also.

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt (hunk 7 above)
Comment (priyatanu) [reply app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt]: thank you. I will check and update

**Comment by BenHenning** [comment]
Hi. As of today, some main reviewers have taken time off for the next few weeks, so it may take a little while before we can look at this PR. We appreciate your patience while some of our team members recharge. We'll be fully returning on 4 January 2021.

**Comment by priyatanu** [comment]
Test1

File: app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt (hunk 7 above)
Comment (priyatanu) [reply app/src/sharedTest/java/org/oppia/android/app/walkthrough/WalkthroughActivityTest.kt]: @rt4914 Please review , test case updated

**Review by Arjupta — APPROVED** [review]
LGTM. @rt4914 PTAL

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Review by BenHenning — APPROVED** [review]
Thanks @priyatanu! Sorry for the late review--this LGTM!

**Pull request merged**